### PR TITLE
Feat/aws AWS S3 provider

### PR DESCRIPTION
**Pull request by andyfelton-equatedigital**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp.Web/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Hi,

I have taken the previous importation (pull request #75) and updated it to be in line with the latest codebase I've also added test cases for cache and file access. I've also moved to the latest master code.

Struggling a bit with GitHub actions. Happy to help out with documentation for this.#

Sorry if I should have updated the old pull request tried and couldn't work out how to do that :-(

Thanks
Andy

### CODE REVIEW

**Comment by andyfelton-equatedigital** [comment]
@JimBobSquarePants I'm noticing the build is failing on some StyleCop things - I'm not seeing this locally. Evenly more oddly one failure is 'AWSS3StorageImageProviderOptions.cs(22,18): error SA1402: File may only contain a single type' which is based on the Azure version and isn't showing an error - I'm probably missing something - can you possibly have a quick look?

**Comment by JimBobSquarePants** [comment]
Will do!

File: src/ImageSharp.Web.Providers.AWS/Providers/AWSS3StorageImageProvider.cs
Comment (JimBobSquarePants) [inline src/ImageSharp.Web.Providers.AWS/Providers/AWSS3StorageImageProvider.cs:57]: We should probably use the same factory code in both the provider and cache for this.

File: src/ImageSharp.Web.Providers.AWS/Providers/AWSS3StorageImageProviderOptions.cs
Comment (JimBobSquarePants) [inline src/ImageSharp.Web.Providers.AWS/Providers/AWSS3StorageImageProviderOptions.cs:45]: What does this comment mean?

**Comment by JimBobSquarePants** [comment]
@andyfelton-equatedigital Looks like I got all the build issues sorted and did a little cleanup also. There was definitely some funkiness going on with the repo. I'd delete your fork after this is merged and recreate it.

**Comment by JimBobSquarePants** [comment]
@andyfelton-equatedigital When you have the opportunity can you temporarily give me write access to your fork? There's something funny going on with Git LFS and while I can push changes to your PR I'm still missing certain permissions to fix the issue.

**Comment by andyfelton-equatedigital** [comment]
@JimBobSquarePants Thanks for all your work on this pull request - above and beyond!

I've invited you as a collaborator on my fork - got to say I was struggling with the LFS files as well. I'll look at your suggestions later today and fix them.

**Comment by andyfelton-equatedigital** [comment]
@JimBobSquarePants I've just improved the comments for the keys/endpoints!

For the factory code would you prefer a shared file with that in?

**Comment by JimBobSquarePants** [comment]
>For the factory code would you prefer a shared file with that in?

Yeah I think so, an internal `AWSS3Factory` class perhaps? I've already normalized the code in the cache so I think the implementation should be based on that.

**Comment by andyfelton-equatedigital** [comment]
> > For the factory code would you prefer a shared file with that in?
> 
> Yeah I think so, an internal `AWSS3Factory` class perhaps? I've already normalized the code in the cache so I think the implementation should be based on that.

That's done!

**Review by JimBobSquarePants — APPROVED** [review]
I think this is ready to go in now. Thanks so much for completing this feature! 👍

**Comment by andyfelton-equatedigital** [comment]
@JimBobSquarePants thanks for your work on it as well - very much appreciated? 

Not pushing but do you have an ETA as to when you'd likely be releasing this just so I can arrange my plan - even happy to pick up an early access version if you have one somewhere?

**Pull request merged**
